modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/inbound/MessagePreProcessor.java

```diff
@@ -268,6 +269,10 @@ private AndesMessage cloneAndesMessageMetadataAndContent(AndesMessage message) {
      */
     private void setMessageID(AndesMessage message) {
         long messageId = idGenerator.getNextId();
+
+        //Tracing message
+        MessageTracer.trace(message, MessageTracer.MESSAGE_ID_MAPPED + " id: " + messageId);
```

Comment:
can't we do like : isTracingEnabled then do this?
 this is in the performance path...